subprojects/docs/src/docs/userguide/troubleshooting/validation_problems.adoc

```diff
@@ -210,5 +210,56 @@ If you use the `@Input` annotation on a file-based property, Gradle wouldn't con
 To fix this problem, you need to tell Gradle if the file property represents an input file, in which case you should annotate it with `@InputFile`, or a directory, in which case it should be annotated with `@InputDirectory`.
 If what you really wanted to say is that the actual file path is an input, then you should return a `String` instead which corresponds to the absolute path of the file.
 
+[[unresolvable_input]]
+== Unresolvable input
 
+This error indicates that a file collection is used as an input of a task, but that resolution failed.
+In general, this shouldn't happen.
+However, when this happens it means that Gradle cannot properly determine the task dependencies.
```

Comment:
It does not. What it means is:
- You added a file collection as a task input
- The task does not have any outputs

So Gradle did consider the file collection for task dependencies, which worked. Though now it wants to resolve the file collection for input/output validation and that doesn't work. Earlier versions of Gradle did not do any validations on input file collections when the task doesn't have any outputs. This is why the task used to pass and now emits the deprecation message. Normally, the task action would evaluate the input file collection and that would fail anyway.

If you only want to use a file collection (or actually the elements of the file collection) to infer dependencies, you can also use `dependsOn` instead of adding the file collection as an input.

Reply:
ok let me try to digest this and come with an explanation for the docs :)